tests/test_hdfs.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import pytest
+import shakedown
+
+import sdk_auth
+import sdk_hosts
+import sdk_install
+
+from tests import utils
+
+
+log = logging.getLogger(__name__)
+GENERIC_HDFS_USER_PRINCIPAL = "hdfs@{realm}".format(realm=sdk_auth.REALM)
+# To do: change when no longer using HDFS stub universe
+HDFS_PACKAGE_NAME='beta-hdfs'
+
+
+@pytest.fixture(scope='module')
+def hdfs_with_kerberos():
+    try:
+        primaries = ["hdfs", "HTTP"]
+        fqdn = "{service_name}.{host_suffix}".format(
+            service_name=utils.HDFS_SERVICE_NAME, host_suffix=sdk_hosts.AUTOIP_HOST_SUFFIX)
+        instances = [
+            "name-0-node",
+            "name-0-zkfc",
+            "name-1-node",
+            "name-1-zkfc",
+            "journal-0-node",
+            "journal-1-node",
+            "journal-2-node",
+            "data-0-node",
+            "data-1-node",
+            "data-2-node",
+        ]
+        principals = []
+        for instance in instances:
+            for primary in primaries:
+                principals.append(
+                    "{primary}/{instance}.{fqdn}@{REALM}".format(
+                        primary=primary,
+                        instance=instance,
+                        fqdn=fqdn,
+                        REALM=sdk_auth.REALM
+                    )
+                )
+        principals.append(GENERIC_HDFS_USER_PRINCIPAL)
+
+        kerberos_env = sdk_auth.KerberosEnvironment()
+        kerberos_env.add_principals(principals)
+        kerberos_env.finalize()
+        service_kerberos_options = {
+            "service": {
+                "kerberos": {
+                    "enabled": True,
+                    "kdc_host_name": kerberos_env.get_host(),
+                    "kdc_host_port": kerberos_env.get_port(),
+                    "keytab_secret": kerberos_env.get_keytab_path(),
+                    "primary": primaries[0],
+                    "primary_http": primaries[1],
+                    "realm": sdk_auth.REALM
+                }
+            }
+        }
+
+        sdk_install.uninstall(HDFS_PACKAGE_NAME, utils.HDFS_SERVICE_NAME)
+        sdk_install.install(
+            HDFS_PACKAGE_NAME,
+            utils.HDFS_SERVICE_NAME,
```

Comment:
Let's move all of the service-specific constants (e.g. `HDFS_SERVICE_NAME`) into the module testing that service?

Reply:
Ok